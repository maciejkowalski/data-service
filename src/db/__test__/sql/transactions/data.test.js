const sql = require('../../../sql/transactions/data');

describe('transactions.data api method', () => {
  it('`one` matches snapshot', () => {
    expect(
      sql.one('2jnH9e2KvEEAiWKHgwocD9arjK3AvBz6DbqwK8GUHdXA')
    ).toMatchSnapshot();
  });

  describe('many matches snapshot with', () => {
    it('no params', () => {
      expect(sql.many()).toMatchSnapshot();
    });

    it('timeStart filter', () => {
      expect(sql.many({ timeStart: new Date('2018-01-01') })).toMatchSnapshot();
    });

    it('timeEnd filter', () => {
      expect(sql.many({ timeEnd: new Date('2018-01-02') })).toMatchSnapshot();
    });

    it('key filter', () => {
      expect(sql.many({ key: 'testboolF' })).toMatchSnapshot();
    });

    it('type and value', () => {
      expect(sql.many({ type: 'integer', value: 11 })).toMatchSnapshot();
    });

    it('sort desc', () => {
      expect(sql.many({ sort: 'desc' })).toMatchSnapshot();
    });

    it('limit', () => {
      expect(sql.many({ limit: 11 })).toMatchSnapshot();
    });
    it('after', () => {
      expect(
        sql.many({
          after: {
            timestamp: new Date('2018-04-07T08:36:52.149Z'),
            id: '23sjEq5zNctBTGqrsapLrPxDkHFM8rJCKF1ti55NRpbF',
            sortDirection: 'asc',
          },
        })
      ).toMatchSnapshot();
    });
  });
});
