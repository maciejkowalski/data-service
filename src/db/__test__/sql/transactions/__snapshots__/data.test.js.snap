// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`transactions.data api method \`one\` matches snapshot 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" = '2jnH9e2KvEEAiWKHgwocD9arjK3AvBz6DbqwK8GUHdXA' order by \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with after 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where (time_stamp, id) > ('2018-04-07 11:36:52.149', '23sjEq5zNctBTGqrsapLrPxDkHFM8rJCKF1ti55NRpbF') group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 100) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with key filter 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"data_key\\" = 'testboolF' group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 100) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with limit 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 11) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with no params 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 100) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with sort desc 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 100) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with timeEnd filter 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"time_stamp\\" < '2018-01-02 03:00:00.000' group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 100) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with timeStart filter 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"time_stamp\\" >= '2018-01-01 03:00:00.000' group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 100) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;

exports[`transactions.data api method many matches snapshot with type and value 1`] = `"select \\"tx_type\\", \\"tx_version\\", \\"height\\", \\"id\\", \\"signature\\", \\"time_stamp\\", \\"proofs\\", \\"sender\\", \\"sender_public_key\\", \\"tx_id\\", \\"data_key\\", \\"data_type\\", \\"data_value_integer\\", \\"data_value_boolean\\", \\"data_value_string\\", \\"data_value_binary\\", \\"position_in_tx\\", fee * 10^(-8) as \\"fee\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"id\\" in (select \\"id\\" from (select \\"id\\", min(\\"time_stamp\\") as \\"time_stamp\\" from \\"txs_12\\" left join \\"txs_12_data\\" on \\"txs_12\\".\\"id\\" = \\"txs_12_data\\".\\"tx_id\\" where \\"data_type\\" = 'integer' and \\"data_value_integer\\" = 11 group by \\"id\\" order by \\"time_stamp\\" desc, \\"id\\" desc limit 100) as \\"filtered\\") order by \\"time_stamp\\" desc, \\"id\\" desc, \\"position_in_tx\\" asc"`;
